src/components/Web3Provider/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { Web3ReactProvider } from '@web3-react/core'
+import { Connector } from '@web3-react/types'
+import {
+  coinbaseWalletHooks,
+  createOrderedConnectors,
+  fortmaticHooks,
+  getConnectorForWallet,
+  gnosisSafe,
+  injectedHooks,
+  MODAL_WALLETS,
+  network,
+  Wallet,
+  walletConnectHooks,
+} from 'connectors'
+import usePrevious from 'hooks/usePrevious'
+import { useEffect, useState } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { updateWalletOverride } from 'state/walletOverride/reducer'
+
+const connect = async (connector: Connector) => {
+  try {
+    if (connector.connectEagerly) {
+      await connector.connectEagerly()
```

Comment:
I thought that `connectEagerly` meant it would attempt connection as soon as it is initialized. Is that wrong?

Reply:
connectEagerly isn’t an argument on the connector anymore, need to handle it manually. any connector that doesn’t show a prompt you just use activate (like network connector)